src/program.cpp

```diff
@@ -564,31 +243,22 @@ value program::to_value() const
     result["target"]  = this->impl->target_name;
     if(not this->impl->target_name.empty())
         result["context"] = this->impl->ctx.to_value();
-    value nodes;
-    print_program(*this, [&](auto ins, const auto& names) {
-        value node;
-        node["output"] = names.at(ins);
-        node["name"]   = ins->name();
-        node["shape"]  = migraphx::to_value(ins->get_shape());
-        if(ins->name() == "@literal")
-            node["literal"] = migraphx::to_value(ins->get_literal());
-        node["operator"] = ins->get_operator().to_value();
-        std::vector<std::string> inputs;
-        std::transform(ins->inputs().begin(),
-                       ins->inputs().end(),
-                       std::back_inserter(inputs),
-                       [&](auto i) { return names.at(i); });
-        node["inputs"] = inputs;
-        nodes.push_back(node);
-    });
-    result["nodes"] = nodes;
+
+    result["modules"] = value::object{};
+    auto& module_val  = result.at("modules");
+    for(auto& m : impl->modules)
+    {
+        module_val[m.first] = m.second.to_value();
+    }
     return result;
 }
+
 void program::from_value(const value& v)
 {
     auto version = v.at("version").to<int>();
     if(version != program_file_version)
-        std::cout << "Warning: Version mismatch" << std::endl;
+        std::cout << "Warning: Program version mismatch" << std::endl;
```

Comment:
This is the format version not program version. Ideally, we should probably throw an error because we usually increment this when the format changes in a non-backwards compatible way.

Reply:
fixed